numpyro/infer/hmc_util.py

```diff
@@ -355,9 +357,10 @@ def warmup_adapter(num_adapt_steps, find_reasonable_step_size=None,
     adaptation_schedule = jnp.array(build_adaptation_schedule(num_adapt_steps))
     num_windows = len(adaptation_schedule)
 
-    def init_fn(z, rng_key, step_size=1.0, inverse_mass_matrix=None, mass_matrix_size=None):
+    def init_fn(z_info, rng_key, step_size=1.0, inverse_mass_matrix=None, mass_matrix_size=None):
         """
-        :param z: Initial position of the integrator.
+        :param tuple z_info: A tuple containing the initial position of the integrator, its
```

Comment:
nit: can we just reuse `IntegratorState` namedtuple for `z_info` and define `r` as None? Also, not important but we can also define a defaults argument in `NamedTuple` and set it to `None`. (See [workaround](https://stackoverflow.com/questions/11351032/namedtuple-and-default-values-for-optional-keyword-arguments) for python 3.6).

Reply:
Thanks, that makes sense to me. I've made the change.